### PR TITLE
chore: remove `template_excludes` in the generation config

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Remove `template_excludes` in the generation config
- Update `README.md`

Follow up of #2792.

### CODE REVIEW

**Review by diegomarquezp — APPROVED** [review]

File: generation_config.yaml [hunk 1]
```diff
@@ -1,19 +1,5 @@
 gapic_generator_version: 2.40.2-SNAPSHOT
 googleapis_commitish: 3d50414a7ff3f0b8ffe8ad7858257396e4f18131
-template_excludes:
```
[the remark this answers is not in the document]
Comment (JoeWang1127) [reply generation_config.yaml:3]: > If we want to create new libraries with empty template_excludes, are we going to create an owlbot.py with empty template_excludes section?

Yes, the exclusions in owlbot.py is empty.

[the remark this answers is not in the document]
Comment (blakeli0) [reply generation_config.yaml:3]: How did we populate it before without hermetic build? Maybe we can populate a default list like this? 
```
    ".github/*",
    ".kokoro/*",
    "samples/*",
    "CODE_OF_CONDUCT.md",
    "CONTRIBUTING.md",
    "LICENSE",
    "SECURITY.md",
    "java.header",
    "license-checks.xml",
    "renovate.json",
    ".gitignore"
```

[the remark this answers is not in the document]
Comment (JoeWang1127) [reply generation_config.yaml:3]: Yes, it's hard coded in the `new-client.py`

File: library_generation/utils/utilities.py
Comment (diegomarquezp) [inline library_generation/utils/utilities.py:1]: I think we still confirm the template excludes behave correctly using the owlbot.py golden test, right?

**Review by diegomarquezp — APPROVED** [review]

File: library_generation/utils/utilities.py
Comment (JoeWang1127) [reply library_generation/utils/utilities.py:1]: Yes, the unit test is not changed.

File: generation_config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeWang1127) [reply generation_config.yaml:3]: I removed `template_exclusions` from the config.

**Pull request merged**